File: onboarder/src/main/java/com/chyrta/onboarder/OnboarderFragment.java
package com.chyrta.onboarder;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.ColorRes;
import android.support.annotation.DrawableRes;
import android.support.annotation.StringRes;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

public class OnboarderFragment extends Fragment {

    private static final String ONBOARDER_PAGE_TITLE = "onboarder_page_title";
    private static final String ONBOARDER_PAGE_TITLE_RES_ID = "onboarder_page_title_res_id";
    private static final String ONBOARDER_PAGE_TITLE_COLOR = "onboarder_page_title_color";
    private static final String ONBOARDER_PAGE_DESCRIPTION = "onboarder_page_description";
    private static final String ONBOARDER_PAGE_DESCRIPTION_RES_ID = "onboarder_page_description_res_id";
    private static final String ONBOARDER_PAGE_DESCRIPTION_COLOR = "onborader_page_description_color";
    private static final String ONBOARDER_PAGE_IMAGE_RES_ID = "onboarder_page_iamge_res_id";

    private String onboarderTitle;
    private String onboarderDescription;
    @StringRes private int onboarderTitleResId;
    @ColorRes private int onboarderTitleColor;
    @StringRes private int onboarderDescriptionResId;
    @ColorRes private int onboarderDescriptionColor;
    @DrawableRes private int onboarderImageResId;

    private View onboarderView;
    private ImageView ivOnboarderImage;
    private TextView tvOnboarderTitle;
    private TextView tvOnboarderDescription;

    public OnboarderFragment() {}

    public static OnboarderFragment newInstance(OnboarderPage page) {
        Bundle args = new Bundle();
        args.putString(ONBOARDER_PAGE_TITLE, page.getTitle());
        args.putString(ONBOARDER_PAGE_DESCRIPTION, page.getDescription());
        args.putInt(ONBOARDER_PAGE_TITLE_RES_ID, page.getTitleResourceId());
        args.putInt(ONBOARDER_PAGE_DESCRIPTION_RES_ID, page.getDescriptionResourceId());
        args.putInt(ONBOARDER_PAGE_TITLE_COLOR, page.getTitleColor());
        args.putInt(ONBOARDER_PAGE_DESCRIPTION_COLOR, page.getDescriptionColor());
        args.putInt(ONBOARDER_PAGE_IMAGE_RES_ID, page.getImageResourceId());
        OnboarderFragment fragment = new OnboarderFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        Bundle bundle = getArguments();

        onboarderTitle = bundle.getString(ONBOARDER_PAGE_TITLE, null);
        onboarderTitleResId = bundle.getInt(ONBOARDER_PAGE_TITLE_RES_ID, 0);
        onboarderTitleColor = bundle.getInt(ONBOARDER_PAGE_TITLE_COLOR, 0);
        onboarderDescription = bundle.getString(ONBOARDER_PAGE_DESCRIPTION, null);
        onboarderDescriptionResId = bundle.getInt(ONBOARDER_PAGE_DESCRIPTION_RES_ID, 0);
        onboarderDescriptionColor = bundle.getInt(ONBOARDER_PAGE_DESCRIPTION_COLOR, 0);
        onboarderImageResId = bundle.getInt(ONBOARDER_PAGE_IMAGE_RES_ID, 0);

        onboarderView = inflater.inflate(R.layout.fragment_onboarder, container, false);
        ivOnboarderImage = (ImageView) onboarderView.findViewById(R.id.iv_onboarder_image);
        tvOnboarderTitle = (TextView) onboarderView.findViewById(R.id.tv_onboarder_title);
        tvOnboarderDescription = (TextView) onboarderView.findViewById(R.id.tv_onboarder_description);

        if (onboarderTitle != null) {
            tvOnboarderTitle.setText(onboarderTitle);
        }

        if (onboarderTitleResId != 0) {
            tvOnboarderTitle.setText(getResources().getString(onboarderTitleResId));
        }

        if (onboarderDescription != null) {
            tvOnboarderDescription.setText(onboarderDescription);
        }

        if (onboarderDescriptionResId != 0) {
            tvOnboarderDescription.setText(getResources().getString(onboarderDescriptionResId));
        }

        if (onboarderTitleColor != 0) {
            Log.v("ONBOARD_TITLE_COLOR", String.valueOf(onboarderTitleColor));
            tvOnboarderTitle.setTextColor(ContextCompat.getColor(getActivity(), onboarderTitleColor));
        }

        if (onboarderDescriptionColor != 0) {
            Log.v("ONBOARD_DESC_COLOR", String.valueOf(onboarderDescriptionColor));
            tvOnboarderDescription.setTextColor(ContextCompat.getColor(getActivity(), onboarderDescriptionColor));
        }

        if (onboarderImageResId != 0) {
            Log.v("ONBOARD_IMAGE", String.valueOf(onboarderImageResId));
            ivOnboarderImage.setImageDrawable(ContextCompat.getDrawable(getActivity(), onboarderImageResId));
        }

        return onboarderView;
    }

    @Override
    public void onDetach() {
        super.onDetach();
    }


}

File: onboarder/src/main/java/com/chyrta/onboarder/OnboarderFragment.kt
package com.chyrta.onboarder

import android.os.Bundle
import android.support.annotation.ColorRes
import android.support.annotation.DrawableRes
import android.support.annotation.StringRes
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.support.v7.content.res.AppCompatResources
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

class OnboarderFragment : Fragment() {

    private var onboarderTitle: String? = null
    private var onboarderDescription: String? = null

    @StringRes private var onboarderTitleResId: Int = 0
    @ColorRes private var onboarderTitleColor: Int = 0
    @StringRes private var onboarderDescriptionResId: Int = 0
    @ColorRes private var onboarderDescriptionColor: Int = 0
    @DrawableRes private var onboarderImageResId: Int = 0

    private var onboarderView: View? = null
    private var ivOnboarderImage: ImageView? = null
    private var tvOnboarderTitle: TextView? = null
    private var tvOnboarderDescription: TextView? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val bundle = arguments

        onboarderTitle = bundle.getString(ONBOARDER_PAGE_TITLE, null)
        onboarderTitleResId = bundle.getInt(ONBOARDER_PAGE_TITLE_RES_ID, 0)
        onboarderTitleColor = bundle.getInt(ONBOARDER_PAGE_TITLE_COLOR, 0)
        onboarderDescription = bundle.getString(ONBOARDER_PAGE_DESCRIPTION, null)
        onboarderDescriptionResId = bundle.getInt(ONBOARDER_PAGE_DESCRIPTION_RES_ID, 0)
        onboarderDescriptionColor = bundle.getInt(ONBOARDER_PAGE_DESCRIPTION_COLOR, 0)
        onboarderImageResId = bundle.getInt(ONBOARDER_PAGE_IMAGE_RES_ID, 0)
        onboarderView = inflater!!.inflate(R.layout.fragment_onboarder, container, false)
        ivOnboarderImage = onboarderView!!.findViewById<View>(R.id.iv_onboarder_image) as ImageView
        tvOnboarderTitle = onboarderView!!.findViewById<View>(R.id.tv_onboarder_title) as TextView
        tvOnboarderDescription = onboarderView!!.findViewById<View>(R.id.tv_onboarder_description) as TextView

        if (onboarderTitle != null) {
            tvOnboarderTitle!!.text = onboarderTitle
        }

        if (onboarderTitleResId != 0) {
            tvOnboarderTitle!!.text = resources.getString(onboarderTitleResId)
        }

        if (onboarderDescription != null) {
            tvOnboarderDescription!!.text = onboarderDescription
        }

        if (onboarderDescriptionResId != 0) {
            tvOnboarderDescription!!.text = resources.getString(onboarderDescriptionResId)
        }

        if (onboarderTitleColor != 0) {
            tvOnboarderTitle!!.setTextColor(ContextCompat.getColor(activity, onboarderTitleColor))
        }

        if (onboarderDescriptionColor != 0) {
            tvOnboarderDescription!!.setTextColor(ContextCompat.getColor(activity, onboarderDescriptionColor))
        }

        if (onboarderImageResId != 0) {
            ivOnboarderImage!!.setImageDrawable(AppCompatResources.getDrawable(activity, onboarderImageResId))
        }

        return onboarderView
    }

    companion object {

        private val ONBOARDER_PAGE_TITLE = "onboarder_page_title"
        private val ONBOARDER_PAGE_TITLE_RES_ID = "onboarder_page_title_res_id"
        private val ONBOARDER_PAGE_TITLE_COLOR = "onboarder_page_title_color"
        private val ONBOARDER_PAGE_DESCRIPTION = "onboarder_page_description"
        private val ONBOARDER_PAGE_DESCRIPTION_RES_ID = "onboarder_page_description_res_id"
        private val ONBOARDER_PAGE_DESCRIPTION_COLOR = "onborader_page_description_color"
        private val ONBOARDER_PAGE_IMAGE_RES_ID = "onboarder_page_iamge_res_id"

        fun newInstance(page: OnboarderPage): OnboarderFragment {
            val args = Bundle()
            args.putString(ONBOARDER_PAGE_TITLE, page.title)
            args.putString(ONBOARDER_PAGE_DESCRIPTION, page.description)
            args.putInt(ONBOARDER_PAGE_TITLE_RES_ID, page.titleResourceId)
            args.putInt(ONBOARDER_PAGE_DESCRIPTION_RES_ID, page.descriptionResourceId)
            args.putInt(ONBOARDER_PAGE_TITLE_COLOR, page.titleColor)
            args.putInt(ONBOARDER_PAGE_DESCRIPTION_COLOR, page.descriptionColor)
            args.putInt(ONBOARDER_PAGE_IMAGE_RES_ID, page.imageResourceId)
            val fragment = OnboarderFragment()
            fragment.arguments = args
            return fragment
        }

    }

}
